libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -495,16 +494,20 @@ bool ContractCompiler::visit(FunctionDefinition const& _function)
 		}
 	//@todo assert that everything is in place now
 
+
 	for (ASTPointer<VariableDeclaration const> const& variable: _function.parameters() + _function.returnParameters())
 		m_context.removeVariable(*variable);
-	for (VariableDeclaration const* localVariable: _function.localVariables())
-		m_context.removeVariable(*localVariable);
 
 	m_context.adjustStackOffset(-(int)c_returnValuesSize);
 
+	CompilerUtils(m_context).popStackSlots(stackSurplus);
```

Comment:
Why is this needed? Why can't this be done right after calling `appendModifierOrFunctionCode();`?